examples/script.py

```diff
@@ -0,0 +1,572 @@
+        time_t createdTimestamp;          // Used internally by the COCO SDKs
+        time_t lastUpdateTimestamp;
+    } coco_std_resource_attribute_info_t;
+
+    typedef struct {
+        coco_std_resource_capability_info_t resourceCapabilityInfo;
+        uint32_t attributeArrCount;
+        coco_std_resource_attribute_info_t *attributeArr;
+    } coco_std_resource_capability_t;
+
+    typedef struct {
+        coco_std_resource_summary_info_t resourceSummaryInfo;
+        uint32_t capabilityArrCount;
+        coco_std_resource_capability_t *capabilityArr;
+    } coco_std_resource_t;
+
+    int32_t coco_device_add_resource(coco_std_resource_t *resourceArr, uint32_t resourceArrCnt,
+                                 uint32_t cmdSeqNum, uint32_t cmdSenderNodeId,
+                                 void *context);
+
+    int32_t  coco_device_resource_attribute_update(coco_std_resource_attribute_info_t *resourceAttribute,
+                                              void *context);
+
+    int32_t coco_device_init_auth(void);
+    """)
+
+
+    #for callbacks
+ffi.cdef("""
+    extern "Python+C" void coco_device_join_nw_status_cb(int32_t status);
+    extern "Python+C" void coco_device_add_res_status_cb(int32_t status, void *context);
+    extern "Python+C" void coco_device_attribute_update_status(int32_t status, void *context);
+    extern "Python+C" void coco_device_data_corruption_cb();
+    extern "Python+C" void coco_device_resource_cmd_cb(coco_std_resource_cmd_t *resourceCmd);
+    """)
+
+
+ #The C library to open
+ffi.set_source("_my_coco",
+"""
+    #include <stdlib.h>
+
+    #include <string.h>
+    #include <stdio.h>
+    #include <unistd.h>
+    #include <time.h>
+    #include <stdbool.h>
+    #include <stdint.h>
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocodevicesdk/coco_device_api.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_api.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_data_storage_control_types.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_data_network_config_types.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_data_illuminance_types.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_data_level_types.h"
+    #include "/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/coco_std_data_on_off_types.h"
+    
+    
+    
+""", 
+    library_dirs = ['/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocodevicesdk/', 
+                    '/mnt/host/tmp/workspace/cocosdk/cocodevicesdk/dist/c/0.65.2/include/cocostandard/'],
+    libraries = ["cocodevicesdk"])
+
+
+ffi.compile(verbose=True)
```

Comment:
move the hard-coded strings to their corresponding files and the load from the files. It makes everything more readable.
QUESTION: If starting from scratch: Should this file be run first in order to set everything up ? Will everything work If this is not run and main is executed directly ?

Reply:
If starting from scratch, this file needs to be run to set everyhting up